eclair-core/src/main/scala/fr/acinq/eclair/io/OpenChannelInterceptor.scala

```diff
@@ -0,0 +1,276 @@
+        validateRemoteChannelType(nonInitiator.temporaryChannelId, nonInitiator.channelFlags, nonInitiator.channelType_opt, nonInitiator.localFeatures, nonInitiator.remoteFeatures) match {
+          case Right(channelType) =>
+            val dualFunded = Features.canUseFeature(nonInitiator.localFeatures, nonInitiator.remoteFeatures, Features.DualFunding)
+            val upfrontShutdownScript = Features.canUseFeature(nonInitiator.localFeatures, nonInitiator.remoteFeatures, Features.UpfrontShutdownScript)
+            val localParams = createLocalParams(nodeParams, nonInitiator.localFeatures, upfrontShutdownScript, channelType, isInitiator = false, dualFunded = dualFunded, nonInitiator.fundingAmount, disableMaxHtlcValueInFlight = false)
+            nodeParams.pluginOpenChannelInterceptor match {
+              case Some(plugin) => queryPlugin(plugin, nonInitiator, localParams, ChannelConfig.standard, channelType)
+              case None =>
+                peer ! SpawnChannelNonInitiator(nonInitiator.open, ChannelConfig.standard, channelType, localParams, nonInitiator.peerConnection.toClassic)
+                waitForRequest()
+            }
+          case Left(ex) =>
+            context.log.warn(s"ignoring remote channel open: ${ex.getMessage}")
+            sendFailure(ChannelExceptionError(ex), nonInitiator)
+            waitForRequest()
+        }
+      case PendingChannelsRateLimiterResponse(PendingChannelsRateLimiter.ChannelRateLimited) =>
+        context.log.warn(s"ignoring remote channel open: rate limited")
+        sendFailure(RateLimitError, nonInitiator)
+        waitForRequest()
+    }
+  }
+
+  private def queryPlugin(plugin: InterceptOpenChannelPlugin, nonInitiator: OpenChannelInterceptor.OpenChannelNonInitiator, localParams: LocalParams, channelConfig: ChannelConfig, channelType: SupportedChannelType): Behavior[Command] =
+    Behaviors.withTimers { timers =>
+      timers.startSingleTimer(PluginTimeout, pluginTimeout)
+      val pluginResponseAdapter = context.messageAdapter[InterceptOpenChannelResponse](PluginOpenChannelResponse)
+      val defaultParams = DefaultParams(localParams.dustLimit, localParams.maxHtlcValueInFlightMsat, localParams.htlcMinimum, localParams.toSelfDelay, localParams.maxAcceptedHtlcs)
+      plugin.openChannelInterceptor ! InterceptOpenChannelReceived(pluginResponseAdapter, nonInitiator, defaultParams)
+      receiveCommandMessage[QueryPluginCommands](context, "queryPlugin") {
+        case PluginOpenChannelResponse(pluginResponse: AcceptOpenChannel) =>
+          val localParams1 = updateLocalParams(localParams, pluginResponse.defaultParams)
+          peer ! SpawnChannelNonInitiator(nonInitiator.open, channelConfig, channelType, localParams1, nonInitiator.peerConnection.toClassic)
+          waitForRequest()
+        case PluginOpenChannelResponse(pluginResponse: RejectOpenChannel) =>
+          sendFailure(PluginRejectedError(pluginResponse.error), nonInitiator)
+          waitForRequest()
```

Comment:
You should cancel the timer in both cases, otherwise this will spam the logs with "received unhandled message: PluginTimeout" errors.

Reply:
Fixed in [3c4137f](https://github.com/ACINQ/eclair/pull/2552/commits/3c4137fdb0bb6c8f712c40f508c9fc0468c13304).